core/templates/dev/head/domain/feedback_thread/FeedbackThreadObjectFactory.ts

```diff
@@ -17,12 +17,23 @@
    thread domain objects.
  */
 
-var oppia = require('AppInit.ts').module;
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-oppia.factory('FeedbackThreadObjectFactory', [function() {
-  var FeedbackThread = function(
-      status, subject, summary, originalAuthorName, lastUpdated, messageCount,
-      stateName, threadId) {
+export class FeedbackThread {
+  status: string;
+  subject: string;
+  summary: string;
+  originalAuthorName: string;
+  lastUpdated: number;
+  messageCount: number;
+  stateName: string;
+  threadId: string;
+  messages: any[];
+  constructor(
+      status: string, subject: string, summary: string,
+      originalAuthorName: string, lastUpdated: number, messageCount: number,
```

Comment:
But if you see FeedbackThreadObjectFactorySpec.ts [here](https://github.com/oppia/oppia/blob/71fdef77b6702b4aff7dc716638df5ab81dd288b/core/templates/dev/head/domain/feedback_thread/FeedbackThreadObjectFactorySpec.ts#L31), it has the value as integer.

Reply:
Can you please verify this with the backend? (currently, our frontend test doesn't match a correct type (As it's not in typescript)

Reply:
Currently, the frontend test doesn't have correct types (As it's not in typescript), make sure to always validate this with the backend values.

Reply:
Actually this is very cumbersome process since we have a vast network of interconnected services and backend. That is why I created #7165 to keep track of this.

Reply:
For *ObjectFactory.ts, I don't think this would be very tough to do as you can find similar backend class with docstring (in most cases).

Reply:
Since I have no knowledge of backend, this would be very hard for me (and strain my timeline). That's why #7165 exists to introduce consistency in this regard and can be done by someone with more reliable knowledge of the backend.

Reply:
Hi @YashJipkate, one easy way to handle this is actually to just manual test (which I hope you're already doing, to avoid breakages). When manual testing, add a console.log to the part that reads the output that's returned from the backend. This will tell you the full structure of the return value and you can confirm it by cross-checking with the relevant values in the backend. You can also do a logging.info of the type in the backend.

This is not much more cumbersome than the manual testing which should be done anyway for this PR. As I mentioned previously, #7165 should only be used for cases where a type is _impossible_ to specify because it depends on something else being converted.

(Btw I suspect this might actually be an int -- of millisecs since the Epoch. Not sure though, and it is definitely worth checking.)

Reply:
Ok changed it, PTAL!

Also, it is a `number` as I checked in the console.